airlock/renderers.py

```diff
@@ -0,0 +1,119 @@
+import csv
+from dataclasses import dataclass
+from email.utils import formatdate
+from functools import cached_property
+from pathlib import Path
+
+from django.http import FileResponse
+from django.template import loader
+from django.template.response import SimpleTemplateResponse
+
+from airlock.business_logic import RequestFile
+
+
+@dataclass
+class Renderer:
+    template = None
+
+    abspath: Path
+    request_file: RequestFile = None
```

Comment:
My instinct here would be to factor this slightly differently and have something:
```py
abspath: Path
relpath: Path
file_cache_id:
```
i.e. pull the conditionals out into one place inside `get_renderer` and avoid the `Renderer` class knowing about `RequestFile`.

I don't have any objection to the way you've done it; only mentioning it for the sake of comparing styles.

Reply:
Have reworked it as per your suggestion, including caching the template lookup at import time, and re-using the already resolved template for the render.